src/Task_21/PersonService.java

```diff
@@ -0,0 +1,109 @@
+package Task_21;
+
+import java.io.*;
+import java.nio.charset.StandardCharsets;
+import java.util.Iterator;
+import java.util.TreeSet;
+
+public class PersonService {
+
+    public static TreeSet<Person> readFromFile(String filePath) {
+
+        TreeSet<Person> people = new TreeSet<>(new FirstComparator());
+        FileInputStream fis = null;
+        InputStreamReader isr = null;
+        BufferedReader reader = null;
+        try {
+            fis = new FileInputStream(filePath);
+            isr = new InputStreamReader(fis, StandardCharsets.UTF_8);
+            reader = new BufferedReader(isr);
+            String strInfo;
+
+            while ((strInfo = reader.readLine()) != null) {
+                String[] arrayInfo = strInfo.split(",");
+
+                Person person = new Person();
+                person.setName(arrayInfo[0]);
+                person.setSurname(arrayInfo[1]);
+                person.setSex(arrayInfo[2]);
+                person.setAge(arrayInfo[3]);
+                people.add(person);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        } finally {
+            try {
+                if (fis != null) {
+                    fis.close();
+                }
+                if (isr != null) {
+                    isr.close();
+                }
+                if (reader != null) {
+                    reader.close();
+                }
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+
+        return (people);
+
+//        //Вывод Tree Set (Проверка)
+//        Iterator<Person> iterator = people.iterator();
+//        System.out.println("Tree set:");
+//        while (iterator.hasNext()) {
+//            System.out.println(iterator.next());
+//        }
+    }
+
+    public static String calculateAmountsOfPeople(TreeSet<Person> setOfPeople) {
+
+        Iterator<Person> iterator = setOfPeople.iterator();
```

Comment:
Why iterator?)
It seems, that simple foreach loop is really easier and understandable